### PR TITLE
builtins: deflake TestSerialNormalizationWithUniqueUnorderedID

### LINKED ISSUE #91858: sql: flake in TestSerialNormalizationWithUniqueUnorderedID
Found here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/7506505?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
    builtins_test.go:151:
          Error Trace:  /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7020/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/sem/builtins/builtins_test_/builtins_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/sem/builtins/builtins_test.go:151
          Error:        "66.02865940312645" is not less than "35.2585"
          Test:         TestSerialNormalizationWithUniqueUnorderedID
          Messages:     chiSquared value of 66.028659 must be less than criticalVal 35.258500 to guarantee distribution is relatively uniform
```

cc @rafiss for triage

### PR DESCRIPTION
**Pull request by ajwerner**
The use of range stats and splits was not very stable and not needed.

Fixes #91858

Release note: None

### CODE REVIEW

**Review by rafiss — APPROVED** [review]

File: pkg/sql/sem/builtins/builtins_test.go [hunk 1]
```diff
@@ -121,16 +121,30 @@ CREATE TABLE t (
 	// Enforce 3 bits worth of range splits in the high order to collect range
```
Comment (rafiss) [inline pkg/sql/sem/builtins/builtins_test.go:121]: does the comment about enforcing range splits still technically apply?

**Review by rafiss — CHANGES_REQUESTED** [review]
thanks for fixing! i think it needs a rebase and unskip for https://github.com/cockroachdb/cockroach/pull/91857

**Review by ajwerner — COMMENTED** [review]
>  i think it needs a rebase and unskip for #91857

Done.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/91868#-:-NHQBk0q6nRwQ1iuEQ9x:brjcs1m)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/sql/sem/builtins/builtins_test.go` line 121 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/91868#-NGrmy29OkdCgOYCME0q-r1-121:-NHQBc8l3GXMuu-pj_Yo:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/8a7541c84bf63394dab0ee0caede339b2319e047/pkg/sql/sem/builtins/builtins_test.go#L121)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

does the comment about enforcing range splits still technically apply?
</blockquote></details>

Done.
___
*[`pkg/sql/sem/builtins/builtins_test.go` line 132 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/91868#-NGrmy29OkdCgOYCME0q-r1-132:-NHQBcqaBaif66qttXv4:b-2bcsdx) ([raw file](https://github.com/cockroachdb/cockroach/blob/8a7541c84bf63394dab0ee0caede339b2319e047/pkg/sql/sem/builtins/builtins_test.go#L132)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

at a high level i get that this is computing a histogram, but want to add some comments about what's up with the bit-shifting and why the numbers `60`, `62`, and `7` were chosen? perhaps explain the `SELECT p AS low, lead(p) OVER () AS high FROM boundaries` window clause a bit too
</blockquote></details>

Done. I generalized the query a bit to take a bits parameter and tried to explain it better.

**Review by rafiss — APPROVED** [review]
lgtm!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/91868#-:-NHUwi9mC49S_6oA9K0Y:b-7nbn20)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Pull request merged**